File: app/src/main/java/com/nuhkoca/offlineapp/data/local/dao/NewsDao.java
package com.nuhkoca.offlineapp.data.local.dao;

import com.nuhkoca.offlineapp.data.local.entity.News;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface NewsDao {

    @Query("SELECT * FROM news")
    LiveData<List<News>> loadNews();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void saveNews(List<News> news);

    @Query("SELECT * FROM news WHERE id=:id")
    LiveData<News> getNews(int id);
}

File: app/src/main/java/com/nuhkoca/offlineapp/data/local/dao/NewsDao.kt
package com.nuhkoca.offlineapp.data.local.dao

import com.nuhkoca.offlineapp.data.local.entity.News

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface NewsDao {

    @Query("SELECT * FROM news")
    fun loadNews(): LiveData<List<News>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun saveNews(news: List<News>)

    @Query("SELECT * FROM news WHERE id=:id")
    fun getNews(id: Int): LiveData<News>
}
